src/Adapter/Cart/QueryHandler/GetCartInformationHandler.php

```diff
@@ -203,18 +203,13 @@ private function extractProductsFromLegacySummary(Cart $cart, array $legacySumma
     {
         $products = [];
         foreach ($legacySummary['products'] as $product) {
-            $products[] = new CartProduct(
-                (int) $product['id_product'],
-                isset($product['id_product_attribute']) ? (int) $product['id_product_attribute'] : 0,
-                $product['name'],
-                isset($product['attributes_small']) ? $product['attributes_small'] : '',
-                $product['reference'],
-                \Tools::ps_round($product['price'], $currency->precision),
-                (int) $product['quantity'],
-                \Tools::ps_round($product['total'], $currency->precision),
-                $this->contextLink->getImageLink($product['link_rewrite'], $product['id_image'], 'small_default'),
-                $this->getProductCustomizedData($cart, $product)
-            );
+            $products[] = $this->buildCartProduct($cart, $currency, $product);
+        }
+
+        if (isset($legacySummary['gift_products'])) {
```

Comment:
```suggestion
        if (isset($legacySummary['gift_products']) && is_array($legacySummary['gift_products'])) {
```
I'm afraid about legacy data